pkg/kubelet/status/status_manager.go

```diff
@@ -714,3 +714,41 @@ func NeedToReconcilePodReadiness(pod *v1.Pod) bool {
 	}
 	return false
 }
+
+//SidecarsStatus contains three bools, whether the pod has sidecars, if the all the sidecars are ready and if the non sidecars are in a waiting state.
+type SidecarsStatus struct {
+	SidecarsPresent   bool
+	SidecarsReady     bool
+	ContainersWaiting bool
```

Comment:
If `ContainersWaiting` is only non-sidecars, could we give it a more descriptive name? (i.e. `NonSidecarContainersWaiting`).

Reply:
yeah will do